packages/blade/src/components/FileUpload/__tests__/FileUpload.test.stories.tsx

```diff
@@ -0,0 +1,437 @@
+      />
+    </Box>
+  );
+};
+
+TestMaxCountFileUpload.play = async ({ canvasElement }) => {
+  const { getByText, queryByText } = within(canvasElement);
+
+  // Create files
+  const blob = new Blob([imageBlob]);
+  const file1 = new File([blob], 'my-image.png', {
+    type: 'image/png',
+  });
+  const file2 = new File([blob], 'my-image2.png', {
+    type: 'image/png',
+  });
+  const file3 = new File([blob], 'my-image3.png', {
+    type: 'image/png',
+  });
+
+  const input = getByText('Drag files here or').closest('div')?.querySelector('input');
+
+  await expect(input).toHaveAttribute('multiple');
+
+  await userEvent.upload(input as HTMLElement, [file1, file2, file3]);
+  await expect(getByText("You can't upload more than 2 files.")).toBeInTheDocument();
+  await expect(queryByText('my-image.png')).not.toBeInTheDocument();
+  await expect(queryByText('my-image2.png')).not.toBeInTheDocument();
+};
+
+export const TestMaxSizeFileUpload: StoryFn<typeof Toast> = (): React.ReactElement => {
+  return (
+    <Box maxWidth="400px">
+      <FileUpload
+        uploadType="single"
+        label="Upload GST certificate"
+        helpText=" .jpg, .jpeg, or .png file only"
+        accept="image/*"
+        isRequired
+        // 10 bytes for testing purpose
+        maxSize={10}
+        necessityIndicator="required"
+      />
+    </Box>
+  );
+};
+
+TestMaxSizeFileUpload.play = async ({ canvasElement }) => {
+  const { getByText } = within(canvasElement);
+
+  // Create a file
+  const blob = new Blob([imageBlob]);
+  const filename = 'my-image.png';
+  const file = new File([blob], filename, {
+    type: 'image/png',
+  });
+
+  const input = getByText('Drag files here or').closest('div')?.querySelector('input');
+
+  await expect(input).not.toHaveAttribute('multiple');
+
+  await userEvent.upload(input as HTMLElement, file);
+  await expect(getByText('File size exceeded.')).toBeInTheDocument();
+};
+
+export const TestAcceptProp: StoryFn<typeof Toast> = (): React.ReactElement => {
+  return (
+    <Box maxWidth="400px">
+      <FileUpload
+        uploadType="single"
+        label="Upload GST certificate"
+        helpText=" .jpg, .jpeg, or .png file only"
+        accept="image/*"
+        isRequired
+        necessityIndicator="required"
+      />
+    </Box>
+  );
+};
+
+TestAcceptProp.play = async ({ canvasElement }) => {
+  const { getByText, queryByText } = within(canvasElement);
+
+  // Create a file
+  const blob = new Blob([pdfBlob]);
+  const filename = 'my-pdf.pdf';
+  const file = new File([blob], filename, {
+    type: 'application/pdf',
+  });
+
+  const input = getByText('Drag files here or').closest('div')?.querySelector('input');
+
+  await expect(input).not.toHaveAttribute('multiple');
+
+  await userEvent.upload(input as HTMLElement, file);
+  await expect(queryByText(filename)).not.toBeInTheDocument();
+};
+
+export const TestFileUploadError: StoryFn<typeof Toast> = (): React.ReactElement => {
```

Comment:
```suggestion
export const TestFileUploadError: StoryFn<typeof FileUpload> = (): React.ReactElement => {
```

Reply:
Mamma miaa! Thank you for catching my copy-pastaaa